.github/workflows/ci.yml

```diff
@@ -11,52 +11,96 @@ concurrency:
   group: ci-${{ github.head_ref || github.ref }}
   cancel-in-progress: true
 
+env:
+  CI: true
+  dist: ember-headless-form/dist
+
 jobs:
-  test:
-    name: 'Tests'
+  install_dependencies:
+    name: Install
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/pnpm
 
+  build:
+    name: Build Tests
+    needs: [install_dependencies]
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Install pnpm
-        uses: pnpm/action-setup@v2
-        with:
-          version: 7
-      - name: Install Node
-        uses: actions/setup-node@v3
-        with:
-          node-version: 18.x
-          cache: pnpm
-      - name: Install Dependencies
-        run: pnpm install --frozen-lockfile
-      - name: Lint
+      - uses: ./.github/actions/pnpm
+      - uses: ./.github/actions/assert-build
+
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    needs: [install_dependencies, build]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/pnpm
+      # To be able to run glint, we need the dist directory
+      - uses: ./.github/actions/download-built-package
+      - name: Lint + Format + Glint
         run: pnpm lint
```

Comment:
not all of our repos are doing an all-in-one lint command, but it's quite nice -- would be good to propagate this to everywhere